packages/block-editor/src/index.js

```diff
@@ -2,11 +2,14 @@
  * WordPress dependencies
  */
 import '@wordpress/blocks';
+import '@wordpress/rich-text';
+import '@wordpress/viewport';
```

Comment:
Do we need this here if `block-editor` never actually `select`s from `'core/viewport'` directly? Same question for it existing in `/editor/src/index.js`

Reply:
In all, I'm not sure we need any of these imports in `/editor/src/index.js`:

- `@wordpress/blocks`
- `@wordpress/nux`
- `@wordpress/rich-text`
- `@wordpress/viewport`

Reply:
Conversely, this file is missing an import for `core/data`. There are multiple references to `select( 'core' ).canUser` . Is that something a block editor should be depending on?

Reply:
I agree with the missing `core/data` and yes, this is not something the block editor should depend on, It's part of the unfinished refactoring of the reusable blocks (and there's another usage for media uploads but should be easy to refactor) but I'm not sure I agree with the removals. My reasoning here is that if we use any module that has a store (whether we call it explicitely with select or implicitely by using components) should be imported.

This is only important for npm consumption as it ensures that these modules are loaded (and their stores) before using the components, selectors... In the WP world, this is enforced by the script dependencies.

You could argue that importing a component will automatically import the store, and that's true probably but I have a preference for expliciteness here because ideally this are removed and replaced by something like:

```js
import { store as blocks } from '@wordpress/blocks';

<RegistryProvider registry={ createRegistry( { 'core/blocks': blocks } ) />
```
(Related to the singleton issue we have)

--- 
I'm going to continue as is for now (added core-data) but we can continue discussing this and we adapt the imports accordingly.